lib/gooddata/lcm/actions/synchronize_users.rb

```diff
@@ -226,74 +230,71 @@ def call(params)
                                            do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
                                            create_non_existing_user_groups: create_non_existing_user_groups)
                     when 'sync_multiple_projects_based_on_custom_id'
-                      new_users.group_by { |u| u[:pid] }.flat_map do |client_id, users|
+                      new_users.group_by { |u| u[:pid] }.flat_pmap do |client_id, users|
                         fail "Client id cannot be empty" if client_id.blank?
-                        project = domain.clients(client_id).project
-                        fail "Client #{client_id} does not have project." unless project
-                        puts "Project #{project.pid} of client #{client_id} will receive #{users.count} users"
-                        project.import_users(users,
-                                             domain: domain,
-                                             whitelists: whitelists,
-                                             ignore_failures: ignore_failures,
-                                             remove_users_from_project: remove_users_from_project,
-                                             do_not_touch_users_that_are_not_mentioned: do_not_touch_users_that_are_not_mentioned,
-                                             create_non_existing_user_groups: create_non_existing_user_groups)
+                        client_project = domain.clients(client_id).project
+                        fail "Client #{client_id} does not have project." unless client_project
```

Comment:
Identical code found in synchronize_users.rb:236, synchronize_users.rb:262 (mass = 72)